ruff_sync.py

```diff
@@ -94,6 +94,26 @@ def _get_cli_parser() -> ArgumentParser:
     return parser
 
 
+def github_url_to_raw_url(url: URL) -> URL:
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (complexity):** Consider inlining the URL transformation logic to simplify the code and enhance maintainability.

While the addition of `github_url_to_raw_url` introduces a clear abstraction for handling GitHub URLs, it also increases the overall complexity of the codebase. Specifically, embedding conditional logic within a utility function for a specific use case can limit its reusability and complicate maintenance. Additionally, intertwining the command-line argument processing with business logic (URL transformation) could make future modifications more challenging.

Consider inlining the URL transformation logic directly where it's used, if this transformation is unique to this context. This approach would keep the logic localized, making the code easier to follow and reducing the need for jumping across different parts of the code to understand specific behaviors. Here's a simplified example of how this could be implemented:

```python
def main() -> None:
    args = PARSER.parse_args()

    # Inline URL transformation for GitHub URLs
    if "github.com" in str(args.upstream) and "/blob/" in str(args.upstream):
        raw_url_str = str(args.upstream).replace("github.com", "raw.githubusercontent.com").replace("/blob/", "/")
        args.upstream = httpx.URL(raw_url_str)

    asyncio.run(
        sync(
            Arguments(
                upstream=args.upstream,
                source=args.source,
                exclude=args.exclude,
            )
        )
    )
```

This keeps the transformation logic close to its usage, simplifying the overall structure and making the code more straightforward to understand and maintain.